test/test_kubeclient.rb

```diff
@@ -92,13 +92,31 @@ def test_api_valid
     stub_request(:get, 'http://localhost:8080/api')
       .to_return(status: 200, body: open_test_json_file('versions_list.json'))
 
+    %w(v1beta1 v1beta2 v1beta3).each do |version|
+      client = Kubeclient::Client.new 'http://localhost:8080/api/', version
+      assert client.api_valid?
+    end
+  end
+
```

Comment:
I think it's also worth to add a test which checks that the logic works well with the default version (when user didn't pass a version)